mlir/tools/level-zero-opt/main.cpp

```diff
@@ -0,0 +1,27 @@
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <mlir/IR/MLIRContext.h>
+#include <mlir/InitAllDialects.h>
+#include <mlir/InitAllPasses.h>
+#include <mlir/Tools/mlir-opt/MlirOptMain.h>
+
+int main(int argc, char **argv) {
+  mlir::registerAllPasses();
+  mlir::DialectRegistry registry;
+  mlir::registerAllDialects(registry);
+  return mlir::failed(MlirOptMain(argc, argv,
```

Comment:
How do your custom dialects and passes get added/registered?

Reply:
We need this tool now for adding more test cases...we can get rid of it once we move to refactor branch

Reply:
What I am trying to say is that the refactor branch already has imex-opt (which is the equivalent of level0-opt) and the infrastructure is ready to actually run your tests in CI. What is missing in refactor that keeps you from issuing the PR against refactor?

Reply:
@Hardcode84 @nbpatel I still do not understand how the MLIR passes get registered. Can you enlighten me?

Reply:
Ok, so it is indeed the static vars. Very weird. Is this the recommend way of doing this?